src/geoserver-publish-imagemosaic/index.js

```diff
@@ -48,9 +57,21 @@ const geoserverPublishImageMosaic = async (workerJob, inputs) => {
       // cf. grc.datastores.createImageMosaicStore
       throw 'Datastore does not exist.'
     }
+
+    // move coverage to coverage store folder to keep file structure clean in data directory
+    const fileName = path.basename(coverageToAdd);
+    const oldPath = coverageToAdd;
+    // the internal geoserver url always starts with 'file:', so we split it and use the second index
+    const coverageStorePath = covStoreObject.coverageStore.url.split(":")[1];
+    // TODO read the geoserver data dir path from rest api
+    const newPath = `/opt/geoserver_data/${coverageStorePath}/${fileName}`;
+
+    // Test if path can be accessed, probably not needed
```

Comment:
i think its not needed as the try / catch will handle issues here, but i guess it does not hurt also. So remove comment or code, you decide